### PR TITLE
Add __sub__ method to Pipeline

### PR DESCRIPTION
**Pull request by WaylonWalker**
This will allow the usage of `-` to subtract nodes off of a pipeline in support of #170.

Main use case avoid unnecessary  network transfer costs by not updating raw data during development with ease.
``` python
# pipeline loaded from project
cars_pipeline = pipeline.only_nodes_with_tags('cars')
raw_pipeline = pipeline.only_nodes_with_tags('raw')
not_raw_cars_pipeline = cars_pipeline - raw_pipeline
```

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
- This contribution is my own, created on my own personal time

## Motivation and Context
#170  Pipeline filtering methods only_nodes, only_nodes_with_tags, only_nodes_with_inputs, only_nodes_with_outputs are great for selecting nodes, and sections of pipelines. Sometimes I want to avoid parts of the pipeline.

## How has this been tested?
manually setup some pipelines and trimmed them.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](/RELEASE.md) file
- [x] Added tests to cover my changes
- [x] Assigned myself to the PR

### CODE REVIEW

File: tests/pipeline/test_pipeline.py [hunk 1]
```diff
@@ -323,6 +323,27 @@ def test_combine(self):
         assert new_pipeline.outputs() == {"output1", "output2"}
         assert {n.name for n in new_pipeline.nodes} == {"a", "b"}
 
+    def test_remove(self):
```
Comment (lorenabalan) [inline tests/pipeline/test_pipeline.py]: Can you add a couple of test cases where you remove a pipeline that has no nodes in common, or respectively only *some* nodes in common? I'm thinking nothing should happen in the first case (p1 - p2 = p1), and some nodes should go away in the second case, like in set algebra.

File: RELEASE.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## Major features and improvements
 * Added a `MatplotlibS3Writer` dataset in `contrib` for saving Matplotlib images to S3.
 * Added a `JSONGCSDataSet` dataset in `contrib` for connecting to Google Cloud Storage.
+* Added `__sub__` method to Pipeline objects for trimming off subsets of a pipeline.
```
Comment (lorenabalan) [inline RELEASE.md]: Minor, maybe clearer/easier to understand for users:
```suggestion
* Pipelines can be deducted with `pipeline1 - pipeline2` .
```

Comment (WaylonWalker) [reply RELEASE.md]: Great suggestion, that is much clearer.

File: tests/pipeline/test_pipeline.py (hunk 1 above)
Comment (WaylonWalker) [reply tests/pipeline/test_pipeline.py]: I have added some tests for empty pipelines, and pipelines with partial intersections.  Thanks for the suggestion this will cover more edge cases.

File: tests/pipeline/test_pipeline.py [hunk 3]
```diff
@@ -323,6 +323,76 @@ def test_combine(self):
...
+        pipeline2 = Pipeline([node(biconcat, ["input", "input1"], "output1", name="a")])
+        new_pipeline = pipeline1 - pipeline2
+        assert new_pipeline.inputs() == set()
+        assert new_pipeline.outputs() == set()
+        assert not new_pipeline.nodes
+
+    def test_invalid_remove(self):
+        p = Pipeline([])
+        pattern = r"unsupported operand type\(s\) for -: 'Pipeline' and 'str'"
+        with pytest.raises(TypeError, match=pattern):
```
Comment (idanov) [inline tests/pipeline/test_pipeline.py]: Isn't that supposed to raise `NotImplemented` exception?

**Comment by lorenabalan** [comment]
@WaylonWalker we'll be happy to merge this in as soon as the conflicts are resolved. :) Thanks again for addressing this.

File: tests/pipeline/test_pipeline.py (hunk 3 above)
Comment (WaylonWalker) [reply tests/pipeline/test_pipeline.py]: It does seem a bit odd at first, I stole the pattern from `invalid_intersection`, `invalid_union`, and `invalid_add`.

Comment (lorenabalan) [reply tests/pipeline/test_pipeline.py]: Found this as a good resource: https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror
